### ISSUE TITLE
All osd down when I update CephCluster spec and add resources.osd

### ISSUE THREAD

**Issue by Negashev** [rook/rook#3162]
**Is this a bug report or feature request?**
* Bug Report

**Deviation from expected behavior:**

osd on device (/dev/mapper/ceph*) down with resorce memory reservation/limit

```bash
2019-05-03 13:10:49.389976 I | rookcmd: starting Rook v1.0.0 with arguments '/rook/rook ceph osd start -- --foreground --id 0 --osd-uuid 324d061c-dfa5-4171-92dc-6e21d010df31 --conf /var/lib/rook/osd0/rook-ceph-system-p-zg5t4.config --cluster ceph --default-log-to-file false'
2019-05-03 13:10:49.390078 I | rookcmd: flag values: --help=false, --log-flush-frequency=5s, --log-level=INFO, --osd-id=0, --osd-store-type=bluestore, --osd-uuid=324d061c-dfa5-4171-92dc-6e21d010df31
2019-05-03 13:10:49.390083 I | op-mon: parsing mon endpoints:
2019-05-03 13:10:49.390086 W | op-mon: ignoring invalid monitor
2019-05-03 13:10:49.390254 I | exec: Running command: stdbuf -oL ceph-volume lvm activate --no-systemd --bluestore 0 324d061c-dfa5-4171-92dc-6e21d010df31
2019-05-03 13:10:49.819861 I | Running command: /bin/mount -t tmpfs tmpfs /var/lib/ceph/osd/ceph-0
2019-05-03 13:10:50.106063 I | Running command: /usr/sbin/restorecon /var/lib/ceph/osd/ceph-0
2019-05-03 13:10:50.404779 I | Running command: /bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0
2019-05-03 13:10:50.695462 I | Running command: /bin/ceph-bluestore-tool --cluster=ceph prime-osd-dir --dev /dev/ceph-66edf104-a9a7-4cc4-a5bf-503f90983a43/osd-data-b78d5f05-3a50-45bf-b65c-a7564b1f8492 --path /var/lib/ceph/osd/ceph-0 --no-mon-config
2019-05-03 13:10:51.031722 I | Running command: /bin/ln -snf /dev/ceph-66edf104-a9a7-4cc4-a5bf-503f90983a43/osd-data-b78d5f05-3a50-45bf-b65c-a7564b1f8492 /var/lib/ceph/osd/ceph-0/block
2019-05-03 13:10:51.323701 I | Running command: /bin/chown -h ceph:ceph /var/lib/ceph/osd/ceph-0/block
2019-05-03 13:10:51.609082 I | Running command: /bin/chown -R ceph:ceph /dev/mapper/ceph--66edf104--a9a7--4cc4--a5bf--503f90983a43-osd--data--b78d5f05--3a50--45bf--b65c--a7564b1f8492
2019-05-03 13:10:51.898247 I | Running command: /bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0
2019-05-03 13:10:52.190122 I | --> ceph-volume lvm activate successful for osd ID: 0
2019-05-03 13:10:52.198089 I | exec: Running command: ceph-osd --foreground --id 0 --osd-uuid 324d061c-dfa5-4171-92dc-6e21d010df31 --conf /var/lib/rook/osd0/rook-ceph-system-p-zg5t4.config --cluster ceph --default-log-to-file false
failed to start osd. Failed to complete '': signal: segmentation fault (core dumped).
```

**Expected behavior:**

osd continue work with reserv and limit memory

**How to reproduce it (minimal and precise):**

**Environment**:
* OS (e.g. from /etc/os-release): 
NAME="Ubuntu"
VERSION="18.04.1 LTS (Bionic Beaver)"
ID=ubuntu
ID_LIKE=debian
PRETTY_NAME="Ubuntu 18.04.1 LTS"
VERSION_ID="18.04"
HOME_URL="https://www.ubuntu.com/"
SUPPORT_URL="https://help.ubuntu.com/"
BUG_REPORT_URL="https://bugs.launchpad.net/ubuntu/"
PRIVACY_POLICY_URL="https://www.ubuntu.com/legal/terms-and-policies/privacy-policy"
VERSION_CODENAME=bionic
UBUNTU_CODENAME=bionic
* Kernel (e.g. `uname -a`):
Linux ceph-2 4.15.0-47-generic #50-Ubuntu SMP Wed Mar 13 10:44:52 UTC 2019 x86_64 x86_64 x86_64 GNU/Linux
* Cloud provider or hardware configuration:
openstack
* Rook version (use `rook version` inside of a Rook Pod):
1.0.0
* Kubernetes version (use `kubectl version`):
Client Version: version.Info{Major:"1", Minor:"10", GitVersion:"v1.10.11", GitCommit:"637c7e288581ee40ab4ca210618a89a555b6e7e9", GitTreeState:"clean", BuildDate:"2018-11-26T14:38:32Z", GoVersion:"go1.9.3", Compiler:"gc", Platform:"windows/amd64"}
Server Version: version.Info{Major:"1", Minor:"13", GitVersion:"v1.13.5", GitCommit:"2166946f41b36dea2c4626f90a77706f426cdea2", GitTreeState:"clean", BuildDate:"2019-03-25T15:19:22Z", GoVersion:"go1.11.5", Compiler:"gc", Platform:"linux/amd64"}
* Kubernetes cluster type (e.g. Tectonic, GKE, OpenShift): RKE
* Storage backend status (e.g. for Ceph use `ceph health` in the [Rook Ceph toolbox](https://rook.io/docs/rook/master/ceph-toolbox.html)):
HEALTH_OK

CephCluster
```yaml
apiVersion: ceph.rook.io/v1
kind: CephCluster
metadata:
  creationTimestamp: 2019-05-03T19:56:20Z
  finalizers:
  - cephcluster.ceph.rook.io
  name: rook-ceph
  namespace: rook-ceph-system
spec:
  cephVersion:
    image: ceph/ceph:v14.2.1-20190430
  dashboard:
    enabled: true
    ssl: false
  dataDirHostPath: /var/lib/rook
  mon:
    allowMultiplePerNode: false
    count: 3
    preferredCount: 0
  network:
    hostNetwork: false
  placement:
    all:
      nodeAffinity:
        requiredDuringSchedulingIgnoredDuringExecution:
          nodeSelectorTerms:
          - matchExpressions:
            - key: nodepool
              operator: In
              values:
              - ceph
      tolerations:
      - key: ceph
        operator: Exists
  rbdMirroring:
    workers: 0
  storage:
    config:
      storeType: bluestore
    devices:
    - FullPath: ""
      config: null
      name: sdb
    - FullPath: ""
      config: null
      name: sdc
    - FullPath: ""
      config: null
      name: sdd
    useAllDevices: true
    useAllNodes: true
status:
  ceph:
    health: HEALTH_OK
    lastChanged: 2019-05-13T15:59:24Z
    lastChecked: 2019-05-13T16:05:27Z
    previousHealth: HEALTH_WARN
  state: Created
```

**Comment by Negashev** [comment]
Duplicate https://github.com/rook/rook/issues/3157

**Comment by galexrt** [comment]
@Negashev #3157 is not a duplicate. You are having `segmentation fault` and #3157 has `failed to fetch mon config (--no-mon-config to skip` and then just fails.

**Comment by travisn** [comment]
@Negashev  The CephCluster CR above doesn't show any limits set. What are the limits and how did you set them?

**Comment by Negashev** [comment]
@travisn osd 4Gb (request/limit)

**Comment by Negashev** [comment]
Repeat with filestore
```yaml
  resources:
    mgr:
      limits:
        memory: 512Mi
      requests:
        memory: 512Mi
    mon:
      limits:
        memory: 1792Mi
      requests:
        memory: 1024Mi
    osd:
      limits:
        memory: 4096Mi
      requests:
        memory: 4096Mi
  storage:
    config:
      storeType: filestore
    directories:
      - path: "/mnt/disk1"
      - path: "/mnt/disk2"
      - path: "/mnt/disk3"
    useAllNodes: true
```

**Comment by leseb** [comment]
Can you send the osd pod log so we can read the coredump? Thanks.

**Comment by Negashev** [comment]
I remove cluster, by append logs from one osd (device mount) in https://github.com/rook/rook/issues/3162#issue-443482665

```bash
2019-05-03 13:10:49.389976 I | rookcmd: starting Rook v1.0.0 with arguments '/rook/rook ceph osd start -- --foreground --id 0 --osd-uuid 324d061c-dfa5-4171-92dc-6e21d010df31 --conf /var/lib/rook/osd0/rook-ceph-system-p-zg5t4.config --cluster ceph --default-log-to-file false'
2019-05-03 13:10:49.390078 I | rookcmd: flag values: --help=false, --log-flush-frequency=5s, --log-level=INFO, --osd-id=0, --osd-store-type=bluestore, --osd-uuid=324d061c-dfa5-4171-92dc-6e21d010df31
2019-05-03 13:10:49.390083 I | op-mon: parsing mon endpoints:
2019-05-03 13:10:49.390086 W | op-mon: ignoring invalid monitor
2019-05-03 13:10:49.390254 I | exec: Running command: stdbuf -oL ceph-volume lvm activate --no-systemd --bluestore 0 324d061c-dfa5-4171-92dc-6e21d010df31
2019-05-03 13:10:49.819861 I | Running command: /bin/mount -t tmpfs tmpfs /var/lib/ceph/osd/ceph-0
2019-05-03 13:10:50.106063 I | Running command: /usr/sbin/restorecon /var/lib/ceph/osd/ceph-0
2019-05-03 13:10:50.404779 I | Running command: /bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0
2019-05-03 13:10:50.695462 I | Running command: /bin/ceph-bluestore-tool --cluster=ceph prime-osd-dir --dev /dev/ceph-66edf104-a9a7-4cc4-a5bf-503f90983a43/osd-data-b78d5f05-3a50-45bf-b65c-a7564b1f8492 --path /var/lib/ceph/osd/ceph-0 --no-mon-config
2019-05-03 13:10:51.031722 I | Running command: /bin/ln -snf /dev/ceph-66edf104-a9a7-4cc4-a5bf-503f90983a43/osd-data-b78d5f05-3a50-45bf-b65c-a7564b1f8492 /var/lib/ceph/osd/ceph-0/block
2019-05-03 13:10:51.323701 I | Running command: /bin/chown -h ceph:ceph /var/lib/ceph/osd/ceph-0/block
2019-05-03 13:10:51.609082 I | Running command: /bin/chown -R ceph:ceph /dev/mapper/ceph--66edf104--a9a7--4cc4--a5bf--503f90983a43-osd--data--b78d5f05--3a50--45bf--b65c--a7564b1f8492
2019-05-03 13:10:51.898247 I | Running command: /bin/chown -R ceph:ceph /var/lib/ceph/osd/ceph-0
2019-05-03 13:10:52.190122 I | --> ceph-volume lvm activate successful for osd ID: 0
2019-05-03 13:10:52.198089 I | exec: Running command: ceph-osd --foreground --id 0 --osd-uuid 324d061c-dfa5-4171-92dc-6e21d010df31 --conf /var/lib/rook/osd0/rook-ceph-system-p-zg5t4.config --cluster ceph --default-log-to-file false
failed to start osd. Failed to complete '': signal: segmentation fault (core dumped).
```

**Comment by leseb** [comment]
Arf, it's too bad we can't catch the dump in the log. I'll look into this in 2 weeks, if someone wants to pick that earlier that fine.

**Comment by leseb** [comment]
Fix proposed here: ceph/ceph#28159, so we have to wait for 14.2.2.

**Comment by leseb** [comment]
Closed via https://github.com/ceph/ceph/pull/28382. The fix will be included in 14.2.2.
